scc/cert/bitset.go

```diff
@@ -62,3 +64,31 @@ func (b BitSet[T]) String() string {
 type unsigned interface {
 	~uint8 | ~uint16 | ~uint32 | ~uint64
 }
+
+// Serialize returns the byte representation of the BitSet.
+func (b *BitSet[T]) Serialize() []byte {
+	return b.mask
+}
+
+// Deserialize restores the BitSet from the given byte representation.
+func (b *BitSet[T]) Deserialize(data []byte) {
+	b.mask = make([]byte, len(data))
+	copy(b.mask, data)
+}
+
+// MarshalJSON converts the BitSet into a JSON-compatible hex string.
+func (b *BitSet[T]) MarshalJSON() ([]byte, error) {
+	return []byte(fmt.Sprintf("\"0x%x\"", b.Serialize())), nil
```

Comment:
For symmetry, this should also use the serialization utility, as the UnmarshalJSON is doing. In particular, this would introduce the special case support for `nil` values.